src/Adapter/Console/ContextLoader.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Console;
+
+use Employee;
+use ConsoleController;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Exception\LogicException;
+use PrestaShop\PrestaShop\Adapter\Shop\Context as ShopContext;
+use PrestaShop\PrestaShop\Core\Console\ContextLoaderInterface;
+
+/**
+ * PrestaShop Context in Console Application
+ */
+final class ContextLoader implements ContextLoaderInterface
+{
+    private $legacyContext;
+    private $shopContext;
+
+    public function __construct(LegacyContext $legacyContext, ShopContext $shopContext, $rootDir)
+    {
+        $this->legacyContext = $legacyContext;
+        $this->shopContext = $shopContext;
+
+        require_once $rootDir . '/../config/config.inc.php';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadConsoleContext(InputInterface $input)
```

Comment:
Is `InputInterface` the right input arg for this command ? It looks like using `HttpRequest` as an input to me 🤔 maybe we should replace it with
```suggestion
    public function loadConsoleContext($employeeId, $shopId, $shopGroupdId)
```

Reply:
If I need to add a new option I will have to break compatibility, and if I use scalar values I will have to validate data in this function when it's not its role (these options should be validated in the "interact" function of the commands.